### ISSUE TITLE
When not using the Admin Key what permissions are needed?

### ISSUE THREAD

**Issue by Joobs** [studioespresso/craft-scout#241]
I am creating a multi environment config, one application with an index per environment. 

Algolia recommends to not use the Admin key, and to make an API key per environment with the correct permissions and each key to only access the relevant index.

But I'm not sure what permissions Scout actually needs to mimic the Admin key?
Details are here: https://www.algolia.com/doc/guides/security/api-keys/#rights-and-restrictions

- Search (search): allowed to perform search operations.
- Browse Index (browse): allowed to retrieve all index data with the browse endpoint.
- Add records (addObject): allowed to add or update a records in the index.
- Delete records (deleteObject): allowed to delete an existing record.
- List indices (listIndexes): allowed to get a list of all existing indices.
- Delete index (deleteIndex): allowed to delete an index.
- Get index settings (settings): allowed to read all index settings.
- Set index settings (editSettings): allowed to update all index settings.
- Use analytics API (analytics): allowed to retrieve data with the Analytics API.
- Use recommendation API (recommendation): allowed to interact with the Recommendation API.
- Use usage API (usage): allowed to retrieve data with the Usage API.
- Access logs (logs): allowed to query the logs.
- Get unretrievable attributes (seeUnretrievableAttributes): allowed to retrieve [unretrievableAttributes](https://www.algolia.com/doc/api-reference/api-parameters/unretrievableAttributes/) for all operations that return records.

Any advice on what permissions Scout needs to work?

**Comment by janhenckens** [comment]
Hey @Joobs, haven't tested this but from your list I'd say there are needed:

Add records (addObject): allowed to add or update a records in the index.
Delete records (deleteObject): allowed to delete an existing record.
List indices (listIndexes): allowed to get a list of all existing indices.
Get index settings (settings): allowed to read all index settings.
Set index settings (editSettings): allowed to update all index settings.

Basically anything to do with adding/changing/removing items in the index, and with managing the index and its settings. Can you give that a try?

**Comment by janhenckens** [comment]
Closing this for now, feel free to reopen is you have any further questions.
